test/channels.spec.ts

```diff
@@ -218,3 +254,17 @@ function trimGuids(object) {
     return object ? object.match(/[^@]+/)[0] : '';
   return object;
 }
+
+function countType(object, type) {
+  if (Array.isArray(object))
```

Comment:
Why not simply look at `_guid` and `.objects` array, can the scope ever have different structure?